src/models/dft/functional_utils.js

```diff
@@ -0,0 +1,163 @@
+import { FC_TYPES, FUNCTIONAL_COMPONENTS } from "./functional_components";
+import { FUNCTIONALS } from "./functional_lookup_table";
+
+/* FUNCTIONAL COMPONENTS */
+
+export function findFunctionalComponent(slug) {
```

Comment:
1. This file has a lot of code and zero docstrings with parameters
2. We don't have use cases for `hasExchangeAndCorrelation` etc. for now, and there's already a lot of code to add.  It's good to have this implementation, but I suggest focusing on the minimal implementation for now to agree on the `0 to 1` path first.